client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx

```diff
@@ -33,6 +33,54 @@ const PatternAssembler: Step = ( { navigation } ) => {
 	const siteId = useSiteIdParam();
 	const siteSlugOrId = siteSlug ? siteSlug : siteId;
 
+	const getPatterns = () =>
+		[ header, ...sections, footer ].filter( ( pattern ) => pattern ) as Pattern[];
+
+	const trackEventPattern = ( {
+		patternType,
+		patternId,
+		eventType,
+	}: {
+		patternType: string;
+		patternId: number;
+		eventType: string;
+	} ) => {
+		let eventName;
+		if ( eventType === 'select' ) {
+			eventName = 'calypso_signup_bcpa_select_pattern_click';
+		} else if ( eventType === 'delete' ) {
+			eventName = 'calypso_signup_bcpa_delete_pattern_click';
+		}
+
+		recordTracksEvent( eventName, {
+			pattern_type: patternType,
+			pattern_id: patternId,
+		} );
+	};
+
+	const trackEventContinue = () => {
+		const patterns = getPatterns();
+		recordTracksEvent( 'calypso_signup_bcpa_donecontinue_click', {
+			pattern_ids: patterns.map( ( { id } ) => id ).join( ',' ),
+			pattern_names: patterns.map( ( { name } ) => name ).join( ',' ),
+			pattern_count: patterns.length,
+		} );
+		patterns.forEach( ( { id, name } ) => {
+			recordTracksEvent( 'calypso_signup_bcpa_pattern_final_select', {
+				pattern_id: id,
+				pattern_name: name,
+			} );
+		} );
+	};
+
+	useEffect( () => {
+		if ( showPatternSelectorType ) {
+			recordTracksEvent( 'calypso_signup_bcpa_show_pattern_selector_click', {
+				pattern_type: showPatternSelectorType,
+			} );
+		}
+	}, [ showPatternSelectorType ] );
```

Comment:
It might be better to create a new function as followed

```js
const showPatternSelector = ( type: string ) => {
  if ( type ) {
    recordTracksEvent( ... );
  }

  setShowPatternSelectorType( type );
}
```

As https://beta.reactjs.org/learn/you-might-not-need-an-effect mentioned, You don’t need Effects to handle user events. In this case, you know exactly what happened when the user wants to select a header, section, or footer. So, the effect is not necessary. 

Here is a similar case:
https://beta.reactjs.org/learn/you-might-not-need-an-effect#sharing-logic-between-event-handlers